subsys/mgmt/hawkbit/shell.c

```diff
@@ -18,13 +18,14 @@ static void cmd_run(const struct shell *sh, size_t argc, char **argv)
 	ARG_UNUSED(argv);
 
 	shell_fprintf(sh, SHELL_INFO, "Starting hawkBit run...\n");
-	switch (hawkbit_probe()) {
+	hawkbit_autohandler(false);
```

Comment:
Are you going to see the same message twice? One from the log and the other one from the shell?

Reply:
yes, if you have the log output activated for that shell. But if you have other log backends activated, like syslog it should also get the logs, independent of the reason why hawkbit was executed.

Reply:
Is there a better way to refactor this mechanism so that it only prints the result to the shell instance that triggered it?

Reply:
IMO it is good, that the autohandler, sending these messages to all log backends, independent of the reason why it was executed (shell or scheduled). It's more like a feature. 
I don't think it is easily possible to omit the logs from the autohandler just for the shell in which it was started, while still sending the log to all other log backends, including the other shells, when they are enabled as log backend.